mmdet/datasets/pipelines/transforms.py

```diff
@@ -2739,3 +2739,268 @@ def __repr__(self):
+    2. Randomly select an image_src, which is resized and cropped
+       and padded similarly.
+    3. Randomly select some objects from the image_src.
+    4. Paste those objects to image_dest directly, for image_dest
+       and image_src have the same size.
+    5. Update object masks of image_dest, for some origin objects
+       may be occluded.
+    6. Generate bboxes from the updated masks and filter some objects
+       which are totally occluded, and adjust bboxes which are partly occluded.
+    7. Append selected bboxes, selected masks, and selected labels,
+       and update results.
+
+    Args:
+        output_size (Sequence[int, int]): [height, width] of output image.
+            Default: [1024, 1024].
+        ratio_range (Sequence[float]): (min_ratio, max_ratio).
+            Default: [0.1, 2.0].
+        flip_ratio (float): The flipping probability.
+            Default: 0.5.
+        min_gt_bbox_wh (tuple[float]): The minimum width and height of bbox.
+            Default: (1e-2, 1e-2)
+        max_paste_objects (int): The maximum number of pasted objects.
+            Default: 100.
+        bbox_occluded_thresh (int): The threshold of occluded bbox.
+            Default: 10.
+        mask_occluded_thresh (int): The threshold of occluded mask.
+            Default: 300.
+    """
+
+    def __init__(
+        self,
+        output_size=[1024, 1024],
+        ratio_range=[0.1, 2.0],
+        flip_ratio=0.5,
+        min_gt_bbox_wh=(1e-2, 1e-2),
+        max_paste_objects=100,
+        bbox_occluded_thresh=10,
+        mask_occluded_thresh=300,
+    ):
+
+        self.output_size = output_size
+        self.ratio_range = ratio_range
+        self.flip_ratio = flip_ratio
+        self.min_gt_bbox_wh = min_gt_bbox_wh
+        self.max_paste_objects = max_paste_objects
+        self.bbox_occluded_thresh = bbox_occluded_thresh
+        self.mask_occluded_thresh = mask_occluded_thresh
+
+    def get_indexes(self, dataset):
+        """Call function to collect indexes.
+
+        Args:
+            dataset (:obj:`MultiImageMixDataset`): The dataset.
+        Returns:
+            list: indexes.
+        """
+        return random.randint(0, len(dataset))
+
+    def __call__(self, results):
+        """Call function to make a copy-paste of image.
+
+        Args:
+            results (dict): Result dict.
+        Returns:
+            dict: Result dict with copy-paste transformed.
+        """
+
+        assert 'mix_results' in results
+        assert len(results['mix_results']
+                   ) == 1, 'Copy-paste only support 2 images now !'
+
+        dest_results = self._scale_jitter(results)
+        src_results = self._scale_jitter(results['mix_results'][0])
+        selected_results = self._select_object(src_results)
+        results = self._copy_paste(dest_results, selected_results)
+        return results
+
+    def _scale_jitter(self, results):
+        img = results['img']
```

Comment:
The final goal is to replace scale_jitter and flip logic by the already implemented transforms. You can make SimpleCopyPaste a wrapper of transforms if it is necessary.

Reply:
This might be a little bit hard to implement for now. We can leave and mark a TODO here.